dynamiqs/sparse.py

```diff
@@ -0,0 +1,121 @@
+from __future__ import annotations
+
+import equinox as eqx
+import jax.numpy as jnp
+from jaxtyping import Array, ArrayLike, Scalar, ScalarLike
+
+from .qarray import QArray
+
+
+class SparseQArray(QArray):
+    diags: Array
+    offsets: tuple[int, ...] = eqx.field(static=True)
```

Comment:
It would be better, but since some methods currently require offsets to be static and Python sets are not hashable objects, I will have to keep it as a tuple for the time being.

Reply:
I agree for the comment and yes the offsets need to be static as there is conditional logic based on their value involved. I am trying to remove this condition and if successful in a timely manner, we can do another PR.